### PR TITLE
:sparkles: App detail drawer: group tags by source and category, add filters

### PR DESCRIPTION
**Pull request by mturley**
* Displays tag categories grouped by source in the drawer
  * Adjusts fetch / indexing logic of tags and tag categories to support grouping by source
* Adds filter dropdowns for source and tag category
  * Enhances FilterToolbar so we can render multiple filter dropdowns side by side
  * Enhances useFilterState so we can have filter values applied with OR logic rather than AND in certain circumstances (multi-selecting tag categories and sources only makes sense with OR because they are mutually exclusive, so otherwise there would only ever be 1 or 0 matches for a filter)

Screenshot 2023-03-07 at 2 28 47 PM

  
Screenshot 2023-03-07 at 2 29 09 PM

### CODE REVIEW

**Comment by ibolton336** [comment]
LGTM -Thanks @mturley

File: client/src/app/pages/applications/components/application-tags/application-tags.tsx [hunk 1]
```diff
@@ -37,20 +62,25 @@ export const ApplicationTags: React.FC<ApplicationTagsProps> = ({
...
       )
         .then((tags) => {
-          const newTagCategories: Map<number, TagCategory> = new Map();
-          const newTags: Map<number, Tag[]> = new Map();
-
-          const tagValidResponses = tags.reduce((prev, current) => {
+          const tagsWithSources = tags.reduce((prev, current, index) => {
             if (current) {
-              return [...prev, current.data];
+              const currentTagWithSource: TagWithSource = {
```
Comment (ibolton336) [inline client/src/app/pages/applications/components/application-tags/application-tags.tsx:67]: +1

File: client/src/app/pages/applications/components/application-tags/application-tags.tsx [hunk 2]
```diff
@@ -37,20 +62,25 @@ export const ApplicationTags: React.FC<ApplicationTagsProps> = ({
...
           }, [] as Tag[]);
-
+          const tagCategoryIds = new Set<number>();
+          tagsWithSources.forEach(
+            (tag) => tag.category?.id && tagCategoryIds.add(tag.category?.id)
+          );
           Promise.all(
-            tagValidResponses.map((tag) =>
-              getTagCategoryById(tag?.category?.id || 0)
+            Array.from(tagCategoryIds).map((tagCategoryId) =>
```
Comment (ibolton336) [inline client/src/app/pages/applications/components/application-tags/application-tags.tsx:82]: So cool!

**Review by ibolton336 — APPROVED** [review]

File: client/src/app/pages/applications/components/application-tags/application-tags.tsx [hunk 3]
```diff
@@ -97,48 +110,134 @@ export const ApplicationTags: React.FC<ApplicationTagsProps> = ({
...
     <ConditionalRender when={isFetching} then={<Spinner isSVG size="md" />}>
-      {Array.from(tagCategories.values())
-        .sort((a, b) => (a.rank || 0) - (b.rank || 0))
-        .map((tagCategory) => {
+      <Toolbar
+        clearAllFilters={() => setFilterValues({})}
+        clearFiltersButtonText={t("actions.clearAllFilters")}
+      >
+        <ToolbarContent className={spacing.p_0}>
+          <ToolbarItem>Filter by:</ToolbarItem>
```
Comment (mturley) [inline client/src/app/pages/applications/components/application-tags/application-tags.tsx:174]: Ah, @ibolton336 I forgot to put this one through translation. I guess we have to revisit a bunch of those eventually anyway.

**Pull request merged**